### ISSUE TITLE
Error 2

### ISSUE THREAD

**Issue by hamid89** [nanoporetech/dorado#21]
Hell guys, 

Hope you are doing well. I am on the configuration step as mentioned in clone and build section 'cmake --build cmake-build --config Release -j' I come across the following error. 

home/hamid/dorado/dorado/utils/sequence_utils.cpp: In function 'float utils::mean_qscore_from_qstring(const string&)':
/home/hamid/dorado/dorado/utils/sequence_utils.cpp:17:10: error: 'transform' is not a member of 'std'
     std::transform(qstring.begin(), qstring.end(), std::back_inserter(scores),
          ^~~~~~~~~
gmake[2]: *** [CMakeFiles/dorado_lib.dir/build.make:258: CMakeFiles/dorado_lib.dir/dorado/utils/sequence_utils.cpp.o] Error 1
gmake[2]: *** Waiting for unfinished jobs....
[ 87%] Linking CXX executable ../../../../../bin/vbz_test
[ 88%] Linking CXX executable ../../../../../bin/vbz_hdf_plugin_test
/usr/bin/ld: /home/hamid/miniconda3/envs/dorado/lib/libhdf5.a(H5CX.o): undefined reference to symbol 'pthread_setspecific@@GLIBC_2.2.5'
//usr/lib64/libpthread.so.0: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
gmake[2]: *** [dorado/3rdparty/hdf_plugins/vbz_plugin/test/CMakeFiles/vbz_hdf_plugin_test.dir/build.make:120: bin/vbz_hdf_plugin_test] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:1258: dorado/3rdparty/hdf_plugins/vbz_plugin/test/CMakeFiles/vbz_hdf_plugin_test.dir/all] Error 2
gmake[1]: *** Waiting for unfinished jobs....
[ 88%] Built target vbz_test
gmake[1]: *** [CMakeFiles/Makefile2:290: CMakeFiles/dorado_lib.dir/all] Error 2
gmake: *** [Makefile:166: all] Error 2

Thanks in advance..

**Comment by hamid89** [comment]
``home/hamid/dorado/dorado/utils/sequence_utils.cpp: In function 'float utils::mean_qscore_from_qstring(const string&)':
/home/hamid/dorado/dorado/utils/sequence_utils.cpp:17:10: error: 'transform' is not a member of 'std'
     std::transform(qstring.begin(), qstring.end(), std::back_inserter(scores),
          ^~~~~~~~~
gmake[2]: *** [CMakeFiles/dorado_lib.dir/build.make:258: CMakeFiles/dorado_lib.dir/dorado/utils/sequence_utils.cpp.o] Error 1
gmake[2]: *** Waiting for unfinished jobs....
[ 87%] Linking CXX executable ../../../../../bin/vbz_test
[ 88%] Linking CXX executable ../../../../../bin/vbz_hdf_plugin_test
/usr/bin/ld: /home/hamid/miniconda3/envs/dorado/lib/libhdf5.a(H5CX.o): undefined reference to symbol 'pthread_setspecific@@GLIBC_2.2.5'
//usr/lib64/libpthread.so.0: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
gmake[2]: *** [dorado/3rdparty/hdf_plugins/vbz_plugin/test/CMakeFiles/vbz_hdf_plugin_test.dir/build.make:120: bin/vbz_hdf_plugin_test] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:1258: dorado/3rdparty/hdf_plugins/vbz_plugin/test/CMakeFiles/vbz_hdf_plugin_test.dir/all] Error 2
gmake[1]: *** Waiting for unfinished jobs....
[ 88%] Built target vbz_test
gmake[1]: *** [CMakeFiles/Makefile2:290: CMakeFiles/dorado_lib.dir/all] Error 2
gmake: *** [Makefile:166: all] Error 2

**Comment by hamid89** [comment]
`home/hamid/dorado/dorado/utils/sequence_utils.cpp: In function 'float utils::mean_qscore_from_qstring(const string&)':
/home/hamid/dorado/dorado/utils/sequence_utils.cpp:17:10: error: 'transform' is not a member of 'std'
     std::transform(qstring.begin(), qstring.end(), std::back_inserter(scores),
          ^~~~~~~~~
gmake[2]: *** [CMakeFiles/dorado_lib.dir/build.make:258: CMakeFiles/dorado_lib.dir/dorado/utils/sequence_utils.cpp.o] Error 1
gmake[2]: *** Waiting for unfinished jobs....
[ 87%] Linking CXX executable ../../../../../bin/vbz_test
[ 88%] Linking CXX executable ../../../../../bin/vbz_hdf_plugin_test
/usr/bin/ld: /home/hamid/miniconda3/envs/dorado/lib/libhdf5.a(H5CX.o): undefined reference to symbol 'pthread_setspecific@@GLIBC_2.2.5'
//usr/lib64/libpthread.so.0: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
gmake[2]: *** [dorado/3rdparty/hdf_plugins/vbz_plugin/test/CMakeFiles/vbz_hdf_plugin_test.dir/build.make:120: bin/vbz_hdf_plugin_test] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:1258: dorado/3rdparty/hdf_plugins/vbz_plugin/test/CMakeFiles/vbz_hdf_plugin_test.dir/all] Error 2
gmake[1]: *** Waiting for unfinished jobs....
[ 88%] Built target vbz_test
gmake[1]: *** [CMakeFiles/Makefile2:290: CMakeFiles/dorado_lib.dir/all] Error 2
gmake: *** [Makefile:166: all] Error 2
`

**Comment by iiSeymour** [comment]
@hamid89 I've just pushed a52fd9efda854e99d47fe47fa7ed857b66681b83 which should fix the first error.

What OS are you building on?

**Comment by iiSeymour** [comment]
It appears the `libhdf5` you are using from `miniconda` is the other problem. Are you able to get `libhdf5` from the system package manager?

**Comment by rdeborja** [comment]
@hamid89 I ran into the same issues as you on our cluster.  I tried compiling `dorado` using dependencies from source and installed from `conda`, nothing worked. I eventually got our sys admin to install the required linux dependencies using `apt install` and everything installed/compiled successfully.  I'm testing it now but will return to the "from source/conda" installation to see if I can get it to work.
